src/binary/Yast.cc

```diff
@@ -166,6 +168,81 @@ void set_ruby_source_location(VALUE file, VALUE lineno)
   YaST::ee.setLinenumber(FIX2INT(lineno));
 }
 
+/**
+ * Is the function name an UI function which returns a symbol?
+ * @param  function_name name of the function
+ * @return true if function_name is one of "UserInput", "TimeoutUserInput"
+ *   or "PollInput"
+ */
+static bool ui_symbol_function(const char *function_name)
+{
+    return strcmp(function_name, "UserInput") == 0  ||
+        strcmp(function_name, "TimeoutUserInput") == 0 ||
+        strcmp(function_name, "PollInput") == 0;
+}
+
+/**
+ * Is the YCP value :debugHotkey symbol?
+ * @param  val YCP value
+ * @return true when val == :debugHotkey
+ */
+static bool is_debug_symbol(YCPValue val)
+{
+    return !val.isNull() && val->isSymbol() &&
+        val->asSymbol()->symbol() == "debugHotkey";
+}
+
+/**
+ * Is the function name an UI function which returns a map?
+ * @param  function_name name of the function
+ * @return true if function_name is "WaitForEvent"
+ */
+static bool ui_hash_function(const char *function_name)
+{
+    return strcmp(function_name, "WaitForEvent") == 0;
+}
+
+/**
+ * Is the YCP value a map containg a debug UI event?
+ * @param  val YCP value
+ * @return true if val is map and map["EventType"] == "DebugEvent" &&
+ *   map["ID"] == :debugHotkey
+ */
+static bool is_debug_event(YCPValue val)
+{
+    // is it a map?
+    if (val.isNull() || !val->isMap())
+        return false;
+
+    YCPMap map = val->asMap();
+
+    YCPValue event_type = map->value(YCPString("EventType"));
+    // is map["EventType"] == "DebugEvent"?
+    if (event_type.isNull() || !event_type->isString() ||
+        event_type->asString()->value() != "DebugEvent")
+        return false;
+
+    YCPValue event_id = map->value(YCPString("ID"));
+    // is map["ID"] == :debugHotkey?
+    return !event_id.isNull() && event_id->isSymbol() &&
+        event_id->asSymbol()->symbol() == "debugHotkey";
+}
+
+/**
+ * Start the Ruby debugger, it calls "Yast::Debugger.start" Ruby code.
+ * See file ../ruby/yast/debugger.rb for more details.
+ */
+static void start_ruby_debugger()
+{
+    y2warning("Starting the Ruby debugger...");
+
+    rb_require("yast/debugger");
+    // call "Yast::Debugger.start"
+    VALUE module = rb_const_get(rb_cObject, rb_intern("Yast"));
+    VALUE klass = rb_const_get(module, rb_intern("Debugger"));
+    rb_funcall(klass, rb_intern("start"), 0);
```

Comment:
Um, it should not crash, the `Yast::Debugger.start` is quite robust (handles missing `byebug` gem internally, etc...). If it by chance crashes then YaST is probably already so broken then a segfault is inevitable anyway...

Invoking the debugger is not an usual action, it's a kind of a corner case, so I'd keep it simple. We can always improve it later when we find a real case when it crashes...